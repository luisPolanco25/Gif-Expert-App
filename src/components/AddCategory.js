import React, {useState} from 'react';
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {
    
    
    const [inputValue, setInputValue] = useState('');

    // handleInputChange
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    // handleSubmit
    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2) {
        setCategories(categories => [inputValue, ...categories]);
        setInputValue('');
        }
    }  

    return (
        <form onSubmit={handleSubmit}>
            <input 
                type="text"
                value={inputValue}
                onChange={handleInputChange} />
        </form>  
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}