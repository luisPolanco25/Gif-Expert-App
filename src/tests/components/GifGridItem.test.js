import { GifGridItem } from "../../components/GifGridItem";
import {shallow} from 'enzyme';
import React from 'react';

describe('Tests in <GifGridItem />', () => {

    const title = 'A title';
    const url = 'https://localhost/something.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    
    test('should match with the snapshot', () => {    

        expect(wrapper).toMatchSnapshot();

    });

    test('should have a paragraph with the title const', () => {
        
        const p = wrapper.find('p').text().trim();

        expect(p).toBe(title);
        
    });

    test('should have an image with an url and an alt', () => {
        
        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

    });
    
    test('should have the class animate__jackInTheBox', () => {
        
        const div = wrapper.find('div');

        expect(div.prop('className').includes('animate__jackInTheBox')).toBe(true);

    });
    
})

