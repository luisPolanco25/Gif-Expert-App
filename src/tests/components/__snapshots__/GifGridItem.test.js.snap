// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGridItem /> should match with the snapshot 1`] = `
<div
  className="card animate__animated animate__jackInTheBox"
>
  <img
    alt="A title"
    src="https://localhost/something.jpg"
  />
  <p>
    A title
  </p>
</div>
`;
